src/app/youtube/components/search-results/search-results.component.ts

```diff
@@ -0,0 +1,81 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { IFilterSettings } from '../../../shared/models/filter-settings.model';
+import { FilterSettingsService } from 'src/app/core/services/filter-settings.service';
+import { YoutubeApiService } from 'src/app/core/services/youtube-api.service';
+import { Router } from '@angular/router';
+import paths from '../../../constants/router.paths';
+import { Store } from '@ngrx/store';
+import { getCollection } from '../../../redux/selectors/collection.selectors';
+import { pushToCollection } from '../../../redux/actions/collection.actions';
+import { Observable, Subscription } from 'rxjs';
+import { ICollectionItem } from 'src/app/redux/state.models';
+
+@Component({
+  selector: 'app-search-results',
+  templateUrl: './search-results.component.html',
+  styleUrls: ['./search-results.component.scss']
+})
+export class SearchResultsComponent implements OnInit, OnDestroy {
+
+  public collection$: Observable<ICollectionItem[]>;
+  public subscriptions: Subscription[] = [];
+  public hasContent: boolean;
+  public filterSettings: IFilterSettings;
+  public isSpinnerShown: boolean = false;
+
+  set subscription(sbc: Subscription) { this.subscriptions.push(sbc); }
+
+  constructor(
+    private filterSettingsService: FilterSettingsService,
+    private youtubeApiService: YoutubeApiService,
+    private router: Router,
+    private store: Store
+  ) { }
+
+  public ngOnInit(): void {
+    this.collection$ = this.store.select(getCollection);
+
+    this.subscription = this.collection$
+        .subscribe(items => {
+          this.hasContent = !!items.length;
+        });
+
+    this.subscription = this.filterSettingsService.getFilterSettingsObservable()
+      .subscribe((filterSettings) => this.filterSettings = filterSettings);
+
+    this.subscription = this.youtubeApiService.loadMoreObs$
```

Comment:
it looks very confusing when you reassign one variable several times but in fact collect subscriptions. Change the name of the setter